.github/workflows/ci.yml

```diff
@@ -196,24 +200,70 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - name: Checkout code
+      - name: Checkout
         uses: actions/checkout@v2
 
-      - name: Determine tag
-        uses: haya14busa/action-cond@v1
-        id: imagetag
+      - name: Repo metadata
+        id: repo
+        uses: actions/github-script@v3
         with:
-          cond: ${{ github.event_name == 'pull_request' }}
-          if_true: ${{ github.sha }}
-          if_false: "master"
+          script: |
+            const repo = await github.repos.get(context.repo)
+            return repo.data
 
-      - name: Build
-        uses: docker/build-push-action@v1
-        with:
-          # username: ${{ secrets.DOCKER_USERNAME }}
-          # password: ${{ secrets.DOCKER_PASSWORD }}
-          repository: banzaicloud/pipeline
-          tags: ${{ steps.imagetag.outputs.value }}
-          add_git_labels: true
-          # push: ${{ github.event_name == 'push' }}
-          push: false
+      - name: Docker build info
+        id: info
+        run: |
+          DOCKER_IMAGE=ghcr.io/banzaicloud/pipeline
+          VERSION=noop
+          if [ "${{ github.event_name }}" = "schedule" ]; then
+            VERSION=nightly
+          elif [[ $GITHUB_REF == refs/tags/* ]]; then
+            VERSION=${GITHUB_REF#refs/tags/}
+          elif [[ $GITHUB_REF == refs/heads/* ]]; then
+            VERSION=$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
+            if [ "${{ github.event.repository.default_branch }}" = "$VERSION" ]; then
+              VERSION=edge
+            fi
+          elif [[ $GITHUB_REF == refs/pull/* ]]; then
+            VERSION=pr-${{ github.event.number }}
+          fi
+          TAGS="${DOCKER_IMAGE}:${VERSION}"
+          if [[ $VERSION =~ ^v?[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
```

Comment:
**Question**: why are we limiting the number of version digits to 3?
Realistically it should be enough (so this is not a hard change request), but there is always a possiblity to use dates as valid semver version which is not matched by this expression.

Also at the GHA [event trigger](https://github.com/banzaicloud/pipeline/pull/3207/files#diff-e9f950f17198d3d5e3122a44230a09b9R10) we defined it unbound to the number of digits.

Reply:
That's a copy-paste issue

Reply:
Fixed